### PR TITLE
[IMP] web: x2many: support `invisible` attr on custom controls

### PR DESCRIPTION
**Pull request by aab-odoo**
This PR adds the support for the invisible modifier on custom controls (create and button) in list view archs. The given expression can use the context (`context.get("...")`) and the parent record (`parent.some_field`) in the case of a list inside an x2many field.

It also defines a new type of control: `<delete/>`. This control supports a single attribute `invisible`, which allows to conditionally enable the delete/remove feature. The attribute's value is a python expression that can use the context, the parent record and fields of the record itself (i.e. the feature can be disabled at a record level).

request for task~4063960

### CODE REVIEW

File: addons/web/static/src/views/list/list_renderer.js [hunk 1]
```diff
@@ -869,6 +869,11 @@ export class ListRenderer extends Component {
         return this.isX2Many && this.canCreate;
     }
 
+    displayDeleteIcon(record) {
+        const deleteControl = this.controls.find((control) => control.type === "delete");
```
Comment (kebeclibre) [inline addons/web/static/src/views/list/list_renderer.js:873]: perf improvement proposal: find the control delete once onWillRender,, then eval it on a by record basis

**Review by kebeclibre — APPROVED** [review]

**Comment by kebeclibre** [comment]
robodoo rebase-ff r+

**Review by aab-odoo — COMMENTED** [review]
robodoo r+

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**